src/Authentication/OAuth20.php

```diff
@@ -101,7 +101,7 @@ function (RequestInterface $request): RequestInterface {
 
                 // Add query params
                 $uri = $request->getUri();
-                $query = UserFunction::build($this->headers, $fnContext);
+                $query = UserFunction::build($this->query, $fnContext);
```

Comment:
Este som tu nasiel jeden preklep, ktory nebol pokryty testami, tak som to pridal aj do testov.